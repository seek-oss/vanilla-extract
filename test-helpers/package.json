{
  "name": "test-helpers",
  "private": true,
  "version": "0.0.0",
  "main": "src/index.ts",
  "bin": "src/bin.ts",
  "author": "SEEK",
  "license": "MIT",
  "dependencies": {
    "@babel/core": "^7.13.10",
    "@types/mini-css-extract-plugin": "^1.2.2",
    "@types/webpack-dev-server": "^3.11.1",
    "@vanilla-extract/esbuild-plugin": "*",
    "@vanilla-extract/webpack-plugin": "*",
    "babel-loader": "^8.2.2",
    "css-loader": "^5.1.3",
    "got": "^11.8.2",
    "html-webpack-plugin": "^5.2.0",
    "mini-css-extract-plugin": "^1.3.9",
    "minimist": "^1.2.5",
    "path-browserify": "^1.0.1",
    "portfinder": "^1.0.28",
    "style-loader": "^2.0.0",
    "webpack": "^5.27.1",
    "webpack-dev-server": "^3.11.2",
    "webpack-merge": "^5.7.3"
  },
  "devDependencies": {
    "@types/minimist": "^1"
  }
}
