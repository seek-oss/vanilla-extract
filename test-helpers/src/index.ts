export * from './startFixture';
export * from './getStylesheet';

export const getTestNodes = (fixture: string) =>
  require(`@fixtures/${fixture}/test-nodes.json`);
