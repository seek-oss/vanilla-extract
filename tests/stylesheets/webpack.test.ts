import { getStylesheet, startFixture, TestServer } from 'test-helpers';

const workerIndex = parseInt(process.env.JEST_WORKER_ID ?? '', 10);

const fixtures = ['unused-modules', 'themed', 'sprinkles', 'features'];

let testCounter = 0;

describe.each(fixtures)('webpack - %s', (fixture) => {
  let server: TestServer;

  beforeAll(async () => {
    const portRange = 100 * workerIndex;
    server = await startFixture(fixture, {
      type: 'mini-css-extract',
      mode: 'production',
      basePort: 12000 + portRange + testCounter++,
    });
  });

  it('should create valid stylesheet', async () => {
    expect(await getStylesheet(server.url)).toMatchSnapshot();
  });

  afterAll(async () => {
    await server.close();
  });
});
