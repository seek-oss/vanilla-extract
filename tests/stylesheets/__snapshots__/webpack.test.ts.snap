// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack - features should create valid stylesheet 1`] = `
"._87f2ru0 {
  height: 50px;
}
._87f2ru0:after {
  display: block;
  content: \\"Below 700px\\";
}
._87f2ru1 {
  background-color: powderblue;
}
._87f2ru1:hover {
  background-color: slategray;
}
._87f2ru2 {
  background-color: powderblue;
}
._87f2ru2:hover {
  background-color: slategray;
}
._87f2ru3 {
  background-color: powderblue;
}
._87f2ru3:hover {
  background-color: slategray;
}
._87f2ru5 {
  color: white;
}
._87f2ru6 {
  background-color: black;
}
body ._87f2ru7 {
  font-size: 24px;
}
._87f2ru8 {
  color: white;
}
._87f2ru9 {
  background-color: black;
}
body ._87f2rua {
  font-size: 24px;
}
@media screen and (min-width: 700px) {
  ._87f2ru0 {
    color: plum;
  }
  ._87f2ru0:after {
    content: \\"Above 700px\\";
  }
}

/*# sourceMappingURL=main.css.map*/"
`;

exports[`webpack - sprinkles should create valid stylesheet 1`] = `
"._3nw5tz2 {
  display: flex;
}
._3nw5tz6 {
  display: none;
}
._3nw5tza {
  display: block;
}
._3nw5tze {
  padding-top: 10px;
}
._3nw5tzi {
  padding-top: 20px;
}
._3nw5tzm {
  --_3nw5tz0: 1;
  background: rgba(255, 0, 0, var(--_3nw5tz0));
}
._3nw5tzq {
  --_3nw5tz0: 1;
}
._3nw5tzu {
  --_3nw5tz0: 0.1;
}
._3nw5tzy {
  --_3nw5tz0: 0.2;
}
._3nw5tz12 {
  --_3nw5tz0: 0.3;
}
._3nw5tz16 {
  --_3nw5tz1: 1;
  color: rgba(255, 0, 0, var(--_3nw5tz1));
}
._3nw5tz17 {
  --_3nw5tz1: 1;
}
._3nw5tz18 {
  --_3nw5tz1: 0.8;
}
body > ._3nw5tz1a {
  background: red;
}
@media screen and (min-width: 768px) {
  ._3nw5tz3 {
    display: flex;
  }
  ._3nw5tz7 {
    display: none;
  }
  ._3nw5tzb {
    display: block;
  }
  ._3nw5tzf {
    padding-top: 10px;
  }
  ._3nw5tzj {
    padding-top: 20px;
  }
  ._3nw5tzn {
    --_3nw5tz0: 1;
    background: rgba(255, 0, 0, var(--_3nw5tz0));
  }
  ._3nw5tzr {
    --_3nw5tz0: 1;
  }
  ._3nw5tzv {
    --_3nw5tz0: 0.1;
  }
  ._3nw5tzz {
    --_3nw5tz0: 0.2;
  }
  ._3nw5tz13 {
    --_3nw5tz0: 0.3;
  }
}
@media screen and (min-width: 1024px) {
  ._3nw5tz4 {
    display: flex;
  }
  ._3nw5tz8 {
    display: none;
  }
  ._3nw5tzc {
    display: block;
  }
  ._3nw5tzg {
    padding-top: 10px;
  }
  ._3nw5tzk {
    padding-top: 20px;
  }
  ._3nw5tzo {
    --_3nw5tz0: 1;
    background: rgba(255, 0, 0, var(--_3nw5tz0));
  }
  ._3nw5tzs {
    --_3nw5tz0: 1;
  }
  ._3nw5tzw {
    --_3nw5tz0: 0.1;
  }
  ._3nw5tz10 {
    --_3nw5tz0: 0.2;
  }
  ._3nw5tz14 {
    --_3nw5tz0: 0.3;
  }
  @supports not (display: grid) {
    [data-dark-mode] ._3nw5tz5 {
      display: flex;
    }
    [data-dark-mode] ._3nw5tz9 {
      display: none;
    }
    [data-dark-mode] ._3nw5tzd {
      display: block;
    }
    [data-dark-mode] ._3nw5tzh {
      padding-top: 10px;
    }
    [data-dark-mode] ._3nw5tzl {
      padding-top: 20px;
    }
    [data-dark-mode] ._3nw5tzp {
      --_3nw5tz0: 1;
      background: rgba(255, 0, 0, var(--_3nw5tz0));
    }
    [data-dark-mode] ._3nw5tzt {
      --_3nw5tz0: 1;
    }
    [data-dark-mode] ._3nw5tzx {
      --_3nw5tz0: 0.1;
    }
    [data-dark-mode] ._3nw5tz11 {
      --_3nw5tz0: 0.2;
    }
    [data-dark-mode] ._3nw5tz15 {
      --_3nw5tz0: 0.3;
    }
  }
}

/*# sourceMappingURL=main.css.map*/"
`;

exports[`webpack - themed should create valid stylesheet 1`] = `
":root, .cdwe0v0 {
  --cdwe0v1: blue;
  --cdwe0v2: white;
  --cdwe0v3: 4px;
  --cdwe0v4: 8px;
  --cdwe0v5: 12px;
}
.cdwe0v6 {
  --cdwe0v1: green;
  --cdwe0v2: white;
  --cdwe0v3: 8px;
  --cdwe0v4: 12px;
  --cdwe0v5: 16px;
}
.cdwe0v7 {
  --cdwe0v1: pink;
  --cdwe0v2: purple;
  --cdwe0v3: 6px;
  --cdwe0v4: 12px;
  --cdwe0v5: 18px;
}
@media screen and (min-width: 768px) {
  .cdwe0v7 {
    --cdwe0v1: purple;
    --cdwe0v2: pink;
  }
}
._5iqv0q0 {
  box-shadow: 0 0 5px red;
}
body {
  background-color: skyblue;
}
@font-face {
  src: local(\\"Impact\\");
  font-family: \\"blj6yb0\\";
}
@font-face {
  src: local(\\"Comic Sans MS\\");
  font-family: MyGlobalComicSans;
}
.blj6yb1 {
  display: flex;
  flex-direction: column;
  gap: var(--cdwe0v4);
  padding: var(--cdwe0v5);
}
.blj6yb2 {
  z-index: 1;
  position: relative;
}
.blj6yb3 {
  font-family: \\"blj6yb0\\";
  background-color: var(--cdwe0v1, \\"THIS FALLBACK VALUE SHOULD NEVER BE USED\\");
  color: var(--cdwe0v2);
  border-radius: 9999px;
}
.cdwe0v6 .cdwe0v0 .blj6yb1 .blj6yb3 {
  font-family: MyGlobalComicSans;
  outline: 5px solid red;
}
body .blj6yb2:after {
  content: 'I am content';
}
html .blj6yb6 {
  opacity: var(--blj6yb4, 0.5);
}
html .blj6yb7 {
  opacity: var(--blj6yb4, var(--blj6yb5, 0.25));
}
@media only screen and (min-width: 500px) {
  .blj6yb1 {
    border: 1px solid var(--cdwe0v1);
  }
  .blj6yb3 {
    padding: var(--cdwe0v3);
  }
}
@media only screen and (min-width: 1000px) {
  .blj6yb3 {
    padding: var(--cdwe0v4);
  }
}

/*# sourceMappingURL=main.css.map*/"
`;

exports[`webpack - unused-modules should create valid stylesheet 1`] = `
"body {
  background-color: lavender;
}
._13dn9qt0 {
  box-sizing: border-box;
}
body {
  border: 5px solid black;
}
._1mtonhh0 {
  display: flex;
}
.plqi3i0 {
  height: 100px;
  width: 100px;
  background: green;
}

/*# sourceMappingURL=main.css.map*/"
`;
